addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+            }
+            this.moduleIpAddress = this.getThing().getProperties().get(HOST).toString();
+            // add the isActuator elements to the boolean array
+            logger.debug("Setting up Konnected Module WebHook");
+            try {
+                webHookServlet.activate(this);
+            } catch (KonnectedWebHookFail e) {
+                logger.trace("there was an error activating the servelet: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+            try {
+                String response = updateKonnectedModule();
+                if (response == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Unable to communicate with Konnected Module confirm settings or readd thing.");
+                } else {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            }
+
+            catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Running dispose()");
+        logger.debug("Releasing Konnected WebHook");
+        webHookServlet.deactivate();
+    }
+
+    /**
+     * This method constructs the payload that will be sent
+     * to the Konnected module via the put request
+     * it adds the appropriate sensors and actuators to the {@link KonnectedModulePayload}
+     * as well as the location of the callback {@link KonnectedJTTPServlet}
+     * and auth_token which can be used for validation
+     *
+     * @return a json settings payload which can be sent to the Konnected Module based on the Thing
+     */
+    private String contructSettingsPayload() {
+        String hostPath = "";
+        try {
+            hostPath = getHostName() + webHookServlet.getPath();
+            logger.debug("The host path is: {}", hostPath);
+        } catch (UnknownHostException e) {
+            logger.debug("Unable to obtain hostname: {}", e.getMessage());
+            return "none";
+        }
+        String authToken = config.authToken;
+        logger.debug("The Auth_Token is: {}", authToken);
+        KonnectedModulePayload payload = new KonnectedModulePayload(authToken, "http://" + hostPath);
+        payload.setBlink(config.blink);
+        payload.setDiscovery(config.discovery);
+        getThing().getChannels().forEach(channel -> {
+            ChannelUID channelId = channel.getUID();
+            if (isLinked(channel.getUID())) {
+                // adds linked channels to list based on last value of Channel ID
+                // which is set to a number
+                // get the zone number in integer form
+                Integer zone = Integer.parseInt(channelId.getId().substring((channelId.getId().length() - 1)));
```

Comment:
Oh oh. The channelId can be ANYTHING - you must not derive any semantics from it. Might you be lacking a "zone" parameter on the channel types?

Reply:
added zone parameter to the channel types and made the check dependent on this parameter. There is now no information derived from channelIds